mlhaiti/accounts/api/views.py

```diff
@@ -0,0 +1,11 @@
+from django.contrib.auth import get_user_model
+
+from rest_framework import generics, permissions
+from mlhaiti.accounts.api.serializers import UserSerializer
+
+User = get_user_model()
+
+class UserCreateView(generics.CreateAPIView):
```

Comment:
We need more control in the registration process and we'll need to return more data in the response.

Reply:
Be more explicit. Which control ? Which data ?

Reply:
For instance, we may need to return a well-formatted response when the registration failed, and we need to be able to return the token and add more data next to it. The way it is, it's good - this is the cbv defined by the framework itself.

Reply:
Did you ask me to use a custom view for that ? Give me these examples of  well-formated data response you want me to provide. I  will do it

Reply:
Ok. Sometime, a user-friendly message would be ok instead of letting Django deal with
Euh. 
`return Response({'error':True, "error_message": "The registration failed because ..... " , 'data': {}},status=HTTP_401_UNAUTHORIZED)`

Basically, I think you could just override the post method for now, just to anticipate further features implementation

Reply:
And some custom response when the registration succeed too. 

`return Response({'error':False, "message": "Created successfully.......", "data":{ .. }, "code": 200},status=HTTP_200_OK)`

Reply:
I found time to day so I'm working on it. I m going to use [this convention](https://github.com/omniti-labs/jsend) for our response formatting data

Reply:
I m thinking using this format: 
```json
{
 "status": "",  //  200,400,.....etc
 "error": "",   //  True, False
 "data": [],    //  data
 "message": ""  //  Success messages
}
```